### PR TITLE
1.12.40 does not have system_config.file/dir_permission

### PR DESCRIPTION
**Pull request by hestolz**

### CODE REVIEW

File: source/ext/patches/fluentd/in_tail.patch [hunk 1]
```diff
@@ -14,8 +14,8 @@
      end
  
      def configure_parser(conf)
-+      @file_perm = system_config.file_permission || FILE_PERMISSION
-+      @dir_perm = system_config.dir_permission || DIR_PERMISSION
```
Comment (NarineM) [inline source/ext/patches/fluentd/in_tail.patch:18]: why this change is needed now? Did ruby fluentd also make this change? Have you tested custom log flow with this change to see if it works ?

Comment (NarineM) [reply source/ext/patches/fluentd/in_tail.patch:18]: What scenarios get enabled with pos fix for in_tail and why we need the patch?

Comment (hestolz) [reply source/ext/patches/fluentd/in_tail.patch:18]: Hi @NarineM , I tested this, the patch applies and I am seeing custom logs successfully. 

> why this change is needed now? Did ruby fluentd also make this change? Have you tested custom log flow with this change to see if the issue is resolved?

This is to solve a customer-related issue (shared over teams, 279165709). In some scenarios you can get 

```
2021-12-21 12:46:15 +0000 [error]: ERROR Tailfilereader crashed due to an unexpected exit --- No such file or directory @ rb_sysopen
```

**Review by NarineM — APPROVED** [review]
